app/src/main/java/com/myvetpath/myvetpath/AddPicturesActivity.java

```diff
@@ -215,31 +316,80 @@ public void onClick(DialogInterface arg0, int arg1) { //open up the gallery
                                 pictureActionIntent,
                                 GALLERY_PICTURE);
 
-
                     }
                 });
 
         myAlertDialog.setNegativeButton("Camera",
                 new DialogInterface.OnClickListener() {
                     public void onClick(DialogInterface arg0, int arg1) { //open up the camera
                         chosen_method = 1;
-                        Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
-                        startActivityForResult(intent, 0);
+
+                        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+                        // Ensure that there's a camera activity to handle the intent
+                        if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
+                            // Create the File where the photo should go
+                            File photoFile = null;
+                            try {
+                                photoFile = createImageFile();
+                            } catch (IOException ex) {
+                                // Error occurred while creating the File
+                            }
+                            // Continue only if the File was successfully created
+                            if (photoFile != null) {
+                                Uri photoURI = FileProvider.getUriForFile(AddPicturesActivity.this,
+                                        "com.example.android.fileprovider",
+                                        photoFile);
+                                takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, photoURI);
+                                startActivityForResult(takePictureIntent, REQUEST_TAKE_PHOTO);
+                            }
+                        }
+
+//                        Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+//                        startActivityForResult(intent, 0);
 
                     }
                 });
         myAlertDialog.show();
     }
 
+
+    private File createImageFile() throws IOException {
+        // Create an image file name
+        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
+        String imageFileName = "JPEG_" + timeStamp + "_";
+        imageNames[selectedImageView] = imageFileName + ".jpg";
+
+
+        File storageDir = getExternalFilesDir(Environment.DIRECTORY_PICTURES);
```

Comment:
It seems like this is storing the image in Pictures, not the Gallery. This isn't necessarily a problem but we should make a note of it in the info activity so users don't get lost.